Add featured block

andysellick opened the pull request:
⚠️ This repo is Continuously Deployed: make sure you [follow the guidance](https://docs.publishing.service.gov.uk/manual/development-pipeline.html#merge-your-own-pull-request) ⚠️

## What
Adds the featured block for the landing page.

Uses code copied from the hero block to provide the text content, including more flexibility than we probably currently need. Might be good to consolidate this code - there's potentially a third block going to use the same approach.

Image handling is also copied from the hero block, using a picture element and screen pixel density detection to display an appropriate image for the current device.

## Why
Required for the landing page.

## Visual changes
Appearance on desktop. The image should be a set height, and it will determine the minimum height of the box and be centred within it. This means that as the screen gets narrower the left and right edges of the image will be deliberately lost.

Note that the image shown below isn't optimised for height, this is just an example.

Full width desktop | Constrained desktop
----------------- | -------------
![Screenshot 2024-10-10 at 11 54 25](https://github.com/user-attachments/assets/c26253e1-04e6-4cc7-a6e1-2fa7565b1500) | ![Screenshot 2024-10-10 at 11 56 30](https://github.com/user-attachments/assets/e31d33b8-0e9c-4d7f-8611-761b0cd766b2)

This has been built like this so that we never end up with a gap below the image. If the text is longer or the user has a higher font size set, the image will still fill the available height.

Normal | Increased text size
------ | ------
![Screenshot 2024-10-10 at 11 56 30](https://github.com/user-attachments/assets/f152eedd-3f6d-4bc2-ba04-5117f57e570b) | ![Screenshot 2024-10-10 at 11 58 24](https://github.com/user-attachments/assets/ce4c3c92-51fa-4f51-b9ea-63ae023d2451)


Appearance on tablet and below.

![Screenshot 2024-10-09 at 10 47 44](https://github.com/user-attachments/assets/cb80f12e-5281-455f-aa0e-37d0f00c80fe)


Trello card: https://trello.com/c/YCUbimh9/50-build-featured

AshGDS reviewed (approved):
Looks good to me 👍  

However, if I'm looking at the latest designs correctly, the aspect ratio of the image we will use looks more like a square? I'm wondering if it's worth using more square examples just to ensure it will look correct when content is added. Here's some examples. Looks nice with a large resolution block, but might look a bit odd if the image we get is smaller.

(I used placehold.co for the example images)

<img width="981" alt="image" src="https://github.com/user-attachments/assets/0be9ac0e-9ad1-4149-a1de-ccaa2f9b51dc">
 
<img width="988" alt="image" src="https://github.com/user-attachments/assets/2d143220-717e-432a-8d9f-1e2dd204b348">

andysellick commented:
@AshGDS I checked with design and they suggested this alternative, which I've implemented and updated the description to explain. Can you take a look?
